include/onnxruntime/core/framework/ml_value.h

```diff
@@ -40,15 +40,17 @@ struct OrtValue {
   }
 
   bool IsAllocated() const {
-    return data_ && type_;
+    return HasElement() && type_;
   }
 
   template <typename T>
   const T& Get() const {
     ORT_ENFORCE(onnxruntime::DataTypeImpl::GetType<T>() == type_, onnxruntime::DataTypeImpl::GetType<T>(), " != ", type_);
+    ORT_ENFORCE(HasElement(), "This optional type ortValue is None");
     return *static_cast<T*>(data_.get());
```

Comment:
nit: confusing error message. not clear what 'optional type' has to do with anything given that's completely external to OrtValue, and I called OrtValue::Get on an instance of OrtValue and get an error message saying 'ortValue is None'. 

may be simpler to just say OrtValue contains no data.

Reply:
Fixed